packages/fxa-settings/src/components/LinkValidator/index.tsx

```diff
@@ -68,7 +71,7 @@ const LinkValidator = <TModel extends ModelDataProvider>({
     linkType === LinkType['reset-password'] &&
     email !== undefined
   ) {
-    return <LinkExpiredResetPassword {...{ email, viewName }} />;
+    return <LinkExpiredResetPassword {...{ email, viewName, integration }} />;
```

Comment:
It's too bad that this ensnares the 'integration' class. It's the only one that does, and it seems like maybe LinkExpiredResetPassword ends up exhibiting a little bit of feature envy. i.e. it wants to send out the reset password link, but has to connect account class and integration class functionality in order to do so.

I've left a comment about abstracting the sending of the reset password email into being just a simple callback function that we pass in. What do you think about this?

Reply:
Sounds great to me. I'm still looking into everywhere I'm passing `integration` to see where I can improve/simplify based on integration requirements, but feel free to call out any others to make sure I get 'em all 👀